pkg/reconciler/clusteringress/clusteringress.go

```diff
@@ -333,6 +336,33 @@ func dedup(strs []string) []string {
 	return unique
 }
 
+func (c *Reconciler) reconcileVirtualServices(ctx context.Context, ci *v1alpha1.ClusterIngress,
+	desired []*v1alpha3.VirtualService) error {
+	// First, create all needed VirtualServices.
+	kept := sets.NewString()
+	for _, d := range desired {
+		if err := c.reconcileVirtualService(ctx, ci, d); err != nil {
+			return err
+		}
+		kept.Insert(d.Name)
+	}
+	// Now, remove the extra ones.
+	vses, err := c.virtualServiceLister.VirtualServices(resources.VirtualServiceNamespace(ci)).List(
+		labels.Set(map[string]string{
+			serving.RouteLabelKey:          ci.Labels[serving.RouteLabelKey],
+			serving.RouteNamespaceLabelKey: ci.Labels[serving.RouteNamespaceLabelKey]}).AsSelector())
+	for _, vs := range vses {
+		n, ns := vs.Name, vs.Namespace
+		if kept.Has(n) {
+			continue
+		}
+		if err = c.SharedClientSet.NetworkingV1alpha3().VirtualServices(ns).Delete(n, &metav1.DeleteOptions{}); err != nil {
+			return err
```

Comment:
do you want to log or wrap the error for more context?

Reply:
Done